### PR TITLE
Feature/market snapshot

### PR DESCRIPTION
**Pull request by BruceChar**
## Context

* Describe the changes made and why they were made.
 new getMarketStatus rpc to update market snapshot data
* Provide links to any issues or designs for this change.

## Were any new external packages added?

* List external packages added and why they were added

### CODE REVIEW

File: packages/app/src/service/job/mm/heiko/index.ts [hunk 1]
```diff
@@ -31,3 +31,9 @@ export async function updateHeikoPositions() {
     await positionUpdate(it.address, it.asset_id)
   }
 }
+
+// TODO: dump by network
+export async function updateHeikoMarketSnapshot() {
+  log.error(`heiko market snapshot`)
```
Comment (zhihuizhangjaden) [inline packages/app/src/service/job/mm/heiko/index.ts:37]: Why error?

File: packages/app/src/service/job/mm/index.ts [hunk 2]
```diff
@@ -1,4 +1,9 @@
 import { updateHeikoPositions } from './heiko'
 import { updateParaMarketSnapshot, updateParaPositions } from './para'
 
-export const MmHourJobTasks = [updateHeikoPositions, updateParaPositions, updateParaMarketSnapshot]
+export const MmHourJobTasks = [
+  updateHeikoPositions,
+  // updateHeikoMarketSnapshot,
```
Comment (zhihuizhangjaden) [inline packages/app/src/service/job/mm/index.ts:6]: This is not needed?

File: packages/app/src/service/job/mm/heiko/index.ts (hunk 1 above)
Comment (BruceChar) [reply packages/app/src/service/job/mm/heiko/index.ts]: hah! is in progress just a local debug log has been deleted!

File: packages/app/src/service/job/mm/index.ts (hunk 2 above)
Comment (BruceChar) [reply packages/app/src/service/job/mm/index.ts]: in progress local dev

**Review by zhihuizhangjaden — APPROVED** [review]
LGTM

File: packages/app/package.json [hunk 3]
```diff
@@ -9,11 +9,12 @@
...
     "test": "jest",
-    "start": "node dist/src/index.js"
+    "start": "node dist/src/index.js",
+    "postinstall": "patch-package"
   },
   "dependencies": {
-    "@parallel-finance/api": "^1.5.2",
-    "@parallel-finance/types": "^1.5.9",
+    "@parallel-finance/api": "^1.7.1",
+    "@parallel-finance/types": "^1.7.1",
```
Comment (crystalynliu) [inline packages/app/package.json:17]: upgrade to 1.7.3

File: packages/app/src/service/query.ts [hunk 4]
```diff
@@ -328,85 +331,95 @@ export class QueryApi {
...
-        utilization_ratio: utilizationRatio,
+        borrow_index: bigIntStr(borrowIndex),
+        utilization_ratio: bigIntStr(utilizationRatio),
+
+        // TODO: this field is not used any more!
         last_accrued_timestamp: lastAccruedTimestamp,
         block_number: blockHeight,
         block_timestamp: timestamp
-      })
+      } as MarketSnapshot)
```
Comment (crystalynliu) [inline packages/app/src/service/query.ts:378]: why need as here?

Comment (BruceChar) [reply packages/app/src/service/query.ts:378]: otherwise TS type check will failed

Comment (crystalynliu) [reply packages/app/src/service/query.ts:378]: So I think the issue is you are doing the wrong thing here, if the ts type cannot be passed, it means something wrong with your code.

Comment (BruceChar) [reply packages/app/src/service/query.ts:378]: It's 'cause the inject entity parameter is a `typeof BaseEntity`, the repository create need a specified Entity object.So typecast here

**Review by xsteadybcgo — CHANGES_REQUESTED** [review]

File: packages/app/src/service/query.ts [hunk 5]
```diff
@@ -383,30 +376,38 @@ export class QueryApi {
...
-    const curBlockHash = (await this.api.rpc.chain.getBlockHash(curBlock)).toString()
-    const timevalue = Number(
-      (await (await this.api.at(curBlockHash)).query.timestamp.now()).toHex()
-    )
-
-    const blockTimestamp = toUtcTimestamp(timevalue)
-    await Promise.all(
-      markets.map(market => {
+    let statusQuery: Promise<any>[] = []
+    markets.map(assetId => statusQuery.push((this.api.rpc as any).loans.getMarketStatus(assetId)))
```
[the remark this answers is not in the document]
Comment (BruceChar) [reply packages/app/src/service/query.ts:392]: what's your point? see where the var used

File: packages/app/src/service/query.ts (hunk 4 above)
Comment (crystalynliu) [reply packages/app/src/service/query.ts:378]: Okay, let us leave it now, but I think we can improve the type definition future.

File: packages/app/src/service/query.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (crystalynliu) [reply packages/app/src/service/query.ts:392]: I think you can use `map` direct to generate statusQuery instead.
```
const statusQuery = markets.map(assetId => this.api.rpc.loans.getMarketStatus(assetId))
```

**Review by xsteadybcgo — APPROVED** [review]

**Pull request merged**